src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/StringSearchValue.cs

```diff
@@ -63,20 +63,21 @@ public void AcceptVisitor(ISearchValueVisitor visitor)
         }
 
         /// <inheritdoc />
-        public int CompareTo(ISupportSortSearchValue supportSortSearchValue, ComparisonRange range)
+        public int CompareTo(ISupportSortSearchValue other, ComparisonRange range)
         {
-            if (supportSortSearchValue == null)
+            if (other == null)
             {
                 throw new ArgumentException("Value to be compared to cannot be null");
             }
 
-            var otherValue = supportSortSearchValue as StringSearchValue;
+            var otherValue = other as StringSearchValue;
             if (otherValue == null)
             {
                 throw new ArgumentException($"Value to be compared should be of type {typeof(StringSearchValue)}");
             }
-
+#pragma warning disable CA1309
```

Comment:
Can we add a comment for why the rule is disabled here?

Reply:
Let me loop @namadabu here.
I actually not sure why we use this comparison, and different one for Equals.

Reply:
The reason we are using `CompareOptions.IgnoreNonSpace | CompareOptions.IgnoreCase` is because we want to do a case and accent insensitive comparison here. This is to be in-line with the collation used in our SQL tables for the StringSearchParam values

---
In reply to: [617931965](https://github.com/microsoft/fhir-server/pull/1858#discussion_r617931965) [](ancestors = 617931965)